.github/workflows/CI.yml

```diff
@@ -104,6 +104,27 @@ jobs:
         run: cargo build --manifest-path glib-macros/Cargo.toml
       - name: "glib-macros: test"
         run: cargo test --manifest-path glib-macros/Cargo.toml
+      # glib-macros in 2 levels dependent
+      - uses: actions/checkout@v2
+        with:
+          repository: fengalin/glib-rs-two-levels-dependent
+          ref: main
+          path: check_2_levels_dependent
+      - name: Extract repository & branch name
+        shell: bash
+        run: |
+          echo "ACTOR=$(echo ${{ github.actor }})" >> $GITHUB_ENV
+          echo "BRANCH_NAME=$(echo ${{ github.head_ref }})" >> $GITHUB_ENV
+      - name: Check two levels glib dependent
+        run: |
+          export SPECS="{ git = \"https://github.com/${{ env.ACTOR }}/gtk-rs-core\", branch = \"${{ env.BRANCH_NAME }}\" }"
+          printf "\n\n[patch.'https://github.com/gtk-rs/gtk-rs-core']\n" >> Cargo.toml
+          echo "glib-sys = ${SPECS}" >> Cargo.toml
+          echo "gobject-sys = ${SPECS}" >> Cargo.toml
+          echo "glib = ${SPECS}" >> Cargo.toml
+          cat Cargo.toml
+          cargo check
```

Comment:
I couldn't find a better way to patch `Cargo.toml` with the PR author's branch. This could be a problem if they didn't use `gtk-rs-core` for their fork.

Reply:
I think we can just have  a new tests directory with subprojects that do some different tests that can't fit in a classic test situation. This way we don't need to do much other than cargo build/cargo check the tests in that directory.

Reply:
Yes, that's the solution we came up with :) https://github.com/gtk-rs/gtk-rs-core/pull/116#pullrequestreview-675242780

Reply:
Oh my bad

Reply:
np :)